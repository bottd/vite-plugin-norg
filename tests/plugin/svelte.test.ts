import { fileURLToPath } from 'node:url';
import { join, dirname } from 'node:path';
import { norgPlugin } from '../../src/plugin/index.js';

const __filename = fileURLToPath(import.meta.url);
const __dirname = dirname(__filename);

describe('Svelte Generator', () => {
  it.each(['basic.norg', 'code-blocks.norg', 'headings.norg', 'images.norg', 'links.norg'])(
    'should generate correct Svelte component for %s',
    async fixture => {
      const fixturePath = join(__dirname, '../fixtures', fixture);
      const sveltePlugin = norgPlugin({ mode: 'svelte', include: ['**/*.norg'] });
      const mockContext = {
        error: (err: Error) => {
          throw err;
        },
        resolve: () => Promise.resolve({ id: fixturePath }),
      };
      const result = await sveltePlugin.load.call(mockContext, fixturePath);
      expect(result).toMatchSnapshot();
    }
  );
});
