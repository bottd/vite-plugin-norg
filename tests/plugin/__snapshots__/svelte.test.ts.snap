// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Svelte Generator > should generate correct Svelte component for basic.norg 1`] = `
"<script>
  const htmlContent = "<h1 id=\\"main-title\\">Main Title</h1>\\n<p>Text with <i>bold</i>, <em>italic</em>, <code>code</code>, and <s>strikethrough</s>.</p><h2 id=\\"lists\\">Lists</h2>\\n<ul><li>Item 1</li></ul><ul><li>Item 2</li></ul><ol><li>Number 1</li></ol><ol><li>Number 2</li></ol><h2 id=\\"link\\">Link</h2>\\n<p><a href=\\"https://example.com\\" target=\\"_blank\\">link text</a></p>";
</script>
<script context="module">
  export const metadata = {"author":"Drake Bott","tags":["\\"test\\", \\"basic\\""],"title":"Basic Norg"};
</script>
{@html htmlContent}"
`;

exports[`Svelte Generator > should generate correct Svelte component for code-blocks.norg 1`] = `
"<script>
  const htmlContent = "<h1 id=\\"code-blocks\\">Code Blocks</h1>\\n<h2 id=\\"javascript\\">JavaScript</h2>\\n<pre class=\\"shiki shiki-themes github-light github-dark\\" style=\\"background-color:#fff;--shiki-dark-bg:#24292e;color:#24292e;--shiki-dark:#e1e4e8\\" tabindex=\\"0\\"><code><span class=\\"line\\"><span style=\\"color:#D73A49;--shiki-dark:#F97583\\">function</span><span style=\\"color:#6F42C1;--shiki-dark:#B392F0\\"> hello</span><span style=\\"color:#24292E;--shiki-dark:#E1E4E8\\">() { </span><span style=\\"color:#D73A49;--shiki-dark:#F97583\\">return</span><span style=\\"color:#032F62;--shiki-dark:#9ECBFF\\"> \\"world\\"</span><span style=\\"color:#24292E;--shiki-dark:#E1E4E8\\">; }</span></span>\\n<span class=\\"line\\"></span></code></pre><h2 id=\\"python\\">Python </h2>\\n<pre class=\\"shiki shiki-themes github-light github-dark\\" style=\\"background-color:#fff;--shiki-dark-bg:#24292e;color:#24292e;--shiki-dark:#e1e4e8\\" tabindex=\\"0\\"><code><span class=\\"line\\"><span style=\\"color:#D73A49;--shiki-dark:#F97583\\">def</span><span style=\\"color:#6F42C1;--shiki-dark:#B392F0\\"> add</span><span style=\\"color:#24292E;--shiki-dark:#E1E4E8\\">(a, b): </span><span style=\\"color:#D73A49;--shiki-dark:#F97583\\">return</span><span style=\\"color:#24292E;--shiki-dark:#E1E4E8\\"> a </span><span style=\\"color:#D73A49;--shiki-dark:#F97583\\">+</span><span style=\\"color:#24292E;--shiki-dark:#E1E4E8\\"> b</span></span>\\n<span class=\\"line\\"></span></code></pre><h2 id=\\"plain\\">Plain</h2>\\n<pre class=\\"shiki shiki-themes github-light github-dark\\" style=\\"background-color:#fff;--shiki-dark-bg:#24292e;color:#24292e;--shiki-dark:#e1e4e8\\" tabindex=\\"0\\"><code><span class=\\"line\\"><span>plain text</span></span>\\n<span class=\\"line\\"><span></span></span></code></pre>";
</script>
<script context="module">
  export const metadata = {"title":"\\"Code Test\\""};
</script>
{@html htmlContent}"
`;

exports[`Svelte Generator > should generate correct Svelte component for headings.norg 1`] = `
"<script>
  const htmlContent = "<h1 id=\\"h1\\">H1</h1>\\n<p>Content 1</p><h2 id=\\"h2\\">H2</h2>\\n<p>Content 2</p><h3 id=\\"h3\\">H3</h3>\\n<p>Content 3</p><h4 id=\\"h4\\">H4</h4>\\n<p>Content 4</p><h5 id=\\"h5\\">H5</h5>\\n<p>Content 5</p><h6 id=\\"h6\\">H6</h6>\\n<p>Content 6</p>";
</script>
<script context="module">
  export const metadata = {"title":"\\"Headings\\""};
</script>
{@html htmlContent}"
`;

exports[`Svelte Generator > should generate correct Svelte component for images.norg 1`] = `
"<script>
  const htmlContent = "<h1 id=\\"images\\">Images</h1>\\n<p>Image test</p><img src=\\"./relative/image.png\\" alt=\\"with alt text\\" />";
</script>
<script context="module">
  export const metadata = {"authors":"Drake Bott","created":"2025-07-10T17:51:58-0600","title":"Images","updated":"2025-07-10T18:00:36-0600","version":"1.1.1"};
</script>
{@html htmlContent}"
`;

exports[`Svelte Generator > should generate correct Svelte component for links.norg 1`] = `
"<script>
  const htmlContent = "<h1 id=\\"link-types\\">Link Types</h1>\\n<h2 id=\\"url-links\\">URL Links</h2>\\n<p><a href=\\"https://example.com\\" target=\\"_blank\\">https://example.com</a> <a href=\\"https://github.com/nvim-neorg\\" target=\\"_blank\\">Neorg GitHub</a></p><h2 id=\\"file-links\\">File Links</h2><h2 id=\\"heading-links\\">Heading Links</h2>\\n<p><a href=\\"#token-text-main-token-whitespace-token-text-heading\\">Token(Text(\\"Main\\"))Token(Whitespace)Token(Text(\\"Heading\\"))</a> <a href=\\"#token-text-sub-token-whitespace-token-text-heading\\">Token(Text(\\"Sub\\"))Token(Whitespace)Token(Text(\\"Heading\\"))</a> <a href=\\"#token-text-introduction\\">Custom link text</a></p><h2 id=\\"mixed-content\\">Mixed Content</h2>\\n<p>This paragraph has a <a href=\\"https://rust-lang.org\\" target=\\"_blank\\">Rust website</a> and .</p>";
</script>
<script context="module">
  export const metadata = {"author":"Drake Bott","title":"Link Test"};
</script>
{@html htmlContent}"
`;
